### PR TITLE
Switch Node.js versions to 16 & 18

### PR DESCRIPTION
**Pull request by Rugvip**
Time to move Node.js support from 14 & 16 to 16 & 18. Along with that we can bump the ECMAScript target to [ES2021](https://node.green/#ES2021)

### CODE REVIEW

**Comment by Rugvip** [comment]
This is blocked by https://github.com/kubernetes-client/javascript/pull/837 right now, since we can't work around the transitive dependency on `openid-client` in a clean way.

**Review by camilaibs — APPROVED** [review]

**Comment by benjdlambert** [comment]
We're waiting for a release of the `@kubernetes/client-node` package if I remember correctly, but it's not released yet.

**Comment by Rugvip** [comment]
We are also waiting for https://github.com/Automattic/node-canvas/issues/2025 if we want to avoid installing the build dependencies for `canvas` into CI

**Comment by Rugvip** [comment]
Doh, waiting for https://github.com/mswjs/msw/issues/1388 too, although that might be the last bit 🤞

**Comment by webark** [comment]
also ran into this issue https://github.com/npm/node-tar/issues/321 where a "readTree" fails due to it not being able to parse a tarball correctly. Adding it here for tracking purposes.

**Comment by webark** [comment]
ahh.. Saw you where just issuing a workaround for the tar issue @Rugvip !

**Comment by webark** [comment]
Node 18 is entering its LTS phase on the 22nd I think? (commenting to prevent closure)

**Comment by webark** [comment]
@Rugvip got it merged and going out with next release! 🥳 https://github.com/npm/node-tar/pull/326 So you should be able to bypass the 4c2d18a hack

**Comment by Rugvip** [comment]
After some more updates it looks like it might just be `node-tar` left until we can update 🎉 . Gonna wait for that as I think it's better to use the proper fix rather than the workaround.

**Comment by webark** [comment]
🎉 6.1.2 was just released!

File: plugins/scaffolder-backend/scripts/build-nunjucks.js [hunk 1]
```diff
@@ -58,7 +58,7 @@ require('esbuild')
     bundle: true,
     format: 'cjs',
     platform: 'node',
-    target: 'node14',
+    target: 'node16',
```
Comment (freben) [inline plugins/scaffolder-backend/scripts/build-nunjucks.js:61]: did you actually perform a rebuild too, but nothing changed?

File: packages/cli/config/tsconfig.json [hunk 2]
```diff
@@ -32,7 +32,7 @@
     "strictNullChecks": true,
     "strictPropertyInitialization": true,
     "stripInternal": true,
-    "target": "ES2019",
+    "target": "ES2021",
```
Comment (freben) [inline packages/cli/config/tsconfig.json:35]: Mmmmm, all of the `??=` and stuff, right? :)

**Review by freben — APPROVED** [review]

**Pull request merged**